pkg/apis/kubedirector/v1beta1/kubedirectorapp_types.go

```diff
@@ -103,13 +103,20 @@ type ServiceEndpoint struct {
 // the same services. At deployment time all role members will receive
 // identical resource assignments.
 type NodeRole struct {
-	ID           string               `json:"id"`
-	Cardinality  string               `json:"cardinality"`
-	ImageRepoTag *string              `json:"imageRepoTag,omitempty"`
-	SetupPackage SetupPackage         `json:"configPackage,omitempty"`
-	PersistDirs  *[]string            `json:"persistDirs,omitempty"`
-	EventList    *[]string            `json:"eventList,omitempty"`
-	MinResources *corev1.ResourceList `json:"minResources,omitempty"`
+	ID            string               `json:"id"`
+	Cardinality   string               `json:"cardinality"`
+	ImageRepoTag  *string              `json:"imageRepoTag,omitempty"`
+	SetupPackage  SetupPackage         `json:"configPackage,omitempty"`
+	PersistDirs   *[]string            `json:"persistDirs,omitempty"`
+	EventList     *[]string            `json:"eventList,omitempty"`
+	MinResources  *corev1.ResourceList `json:"minResources,omitempty"`
+	ContainerSpec *ContainerSpec       `json:"containerSpec,omitempty"`
+}
+
+//ContainerSpec comments
+type ContainerSpec struct {
+	Stdin *bool `json:"stdin,omitempty"`
+	Tty   *bool `json:"tty,omitempty"`
 }
```

Comment:
Any reason to use pointers here?  Do you need to represent a null value here?

Reply:
one of them could be empty right

Reply:
It depends on if you need to differentiate "empty" from "false".

Reply:
Bah, logged in as wrong account again.

Reply:
It depends on if you need to differentiate "empty" from False.